test/verify/check-networking-firewall

```diff
@@ -177,6 +178,66 @@ class TestFirewall(MachineCase):
             b.click("caption button.btn-primary")
             b.wait_present("#cockpit_modal_dialog li input[data-id='empty']")
 
+    @skipImage("Custom ports were introduced in 191", "rhel-7-6-distropkg")
+    def testAddCustomServices(self):
+        b = self.browser
+
+        self.login_and_go("/network/firewall")
+
+        # click on the "Add Services" button
+        b.wait_present("caption button.btn-primary:enabled")
+
+        def open_dialog():
+            b.click("caption button.btn-primary")
+            b.wait_present("#add-services-dialog input[value='ports']")
+            b.click("#add-services-dialog input[value='ports']")
+            b.wait_present("#tcp-ports")
+            b.wait_visible("#tcp-ports")
+
+            time.sleep(1) # We need some time before /etc/services is parsed
+
+        def set_field(sel, val, expected):
+            b.set_input_text(sel, val)
+            b.wait_val("#service-name", expected)
+
+        def check_error(text):
+            b.wait_present(".error-msg:contains({0})".format(text))
+            b.wait_visible(".error-msg:contains({0})".format(text))
+
+        def save(ajde, name, tcp, udp):
+            b.click("button:contains(Add Ports)")
+            b.wait_not_present("#add-services-dialog")
+            line_sel = "tr[data-row-id='{0}']".format(ajde)
+            b.wait_present(line_sel)
+            b.wait_in_text(line_sel, "".join([name, tcp, udp]))
+
+        open_dialog()
+        set_field("#tcp-ports", "80", "WorldWideWeb HTTP")
+        set_field("#tcp-ports", "", "")
+        set_field("#tcp-ports", "80,7", "http, echo")
+        set_field("#udp-ports", "194", "http, echo, irc")
+        set_field("#udp-ports", "194,   50000", "http, echo, irc, etc.")
+        set_field("#tcp-ports", "", "irc, 50000")
+        set_field("#tcp-ports", "https", "https, irc, 50000")
+        save("custom--https-irc-50000", "https, irc, 50000", "443", "194, 50000")
+
+        open_dialog()
+        set_field("#tcp-ports", "80", "WorldWideWeb HTTP")
+        b.set_input_text("#service-name", "I am persistent")
+        b.set_input_text("#tcp-ports", "7")
+        time.sleep(5) # We need validate that the service name did not changed
```

Comment:
This is ugly, but I don't have a better idea off the top of my head. But "need *to* validate … did not change".